index.ts

```diff
@@ -16,57 +15,36 @@ export interface PatternsPluginInput extends PluginInput {
   config: PatternsInputs;
 }
 
-const filterEvents = (
-  events: PluginEvent[],
-  config: PatternsInputs
-): PluginEvent[] => {
-  if (!config.allowedEventTypes) {
-    return events;
-  }
-
-  let allowedEventTypes = config.allowedEventTypes.split(",");
-  allowedEventTypes = allowedEventTypes.map((eventType) => eventType.trim());
-
-  const allowedEventTypesSet = new Set(allowedEventTypes);
-
-  let filteredEvents = events.filter((event) =>
-    allowedEventTypesSet.has(event.event)
-  );
-
-  return filteredEvents;
-};
-
 // Plugin method that runs on plugin load
 //@ts-ignore
-export async function setupPlugin({ config }: Meta<PatternsPluginInput>) {
-  console.log("Loaded Patterns app.");
+export async function setupPlugin({ config, global }: Meta<PatternsPluginInput>) {
+  if (config.allowedEventTypes) {
+    let allowedEventTypes = config.allowedEventTypes.split(",");
+    allowedEventTypes = allowedEventTypes.map((eventType: string) => eventType.trim());
+    global.allowedEventTypesSet = new Set(allowedEventTypes);
+  }
 }
 
 // Plugin method to export events
-export const exportEvents: Plugin<PatternsPluginInput>["exportEvents"] = async (
-  events: PluginEvent[],
-  { config }: Meta<PatternsPluginInput>
+export const onEvent: Plugin<PatternsPluginInput>["onEvent"] = async (
+  event,
+  { config, global }: Meta<PatternsPluginInput>
 ) => {
-  
-  let filteredEvents = filterEvents(events, config);
-  console.log(
-    `Exporting events to Patterns webhook... ${filteredEvents.length}/${events.length} events`
-  );
-  
-  if (!filteredEvents.length) {
-    return;
+  if (global.allowedEventTypesSet) {
+    if (!global.allowedEventTypesSet.has(event.event)) {
+      return
+    }
   }
   
   let response: Response;
   response = await fetch(config.webhookUrl, {
     method: "POST",
     headers: { "Content-Type": "application/json" },
-    body: JSON.stringify(filteredEvents),
+    body: JSON.stringify(event),
```

Comment:
So it used to post an array of objects and now we're posting a single object, I assume we know that's OK per their API?